src/solvers/lowering/byte_operators.cpp

```diff
@@ -711,16 +711,18 @@ exprt lower_byte_extract(const byte_extract_exprt &src, const namespacet &ns)
   const bool little_endian = src.id() == ID_byte_extract_little_endian;
 
   // determine an upper bound of the number of bytes we might need
-  exprt upper_bound=size_of_expr(src.type(), ns);
-  if(upper_bound.is_not_nil())
-    upper_bound = simplify_expr(
+  auto upper_bound_opt = size_of_expr(src.type(), ns);
+  if(upper_bound_opt.has_value())
+    upper_bound_opt = simplify_expr(
       plus_exprt(
-        upper_bound,
-        typecast_exprt::conditional_cast(src.offset(), upper_bound.type())),
+        upper_bound_opt.value(),
+        typecast_exprt::conditional_cast(
+          src.offset(), upper_bound_opt.value().type())),
       ns);
 
   const auto lower_bound_or_nullopt = numeric_cast<mp_integer>(src.offset());
-  const auto upper_bound_or_nullopt = numeric_cast<mp_integer>(upper_bound);
+  const auto upper_bound_or_nullopt =
```

Comment:
Other things like this are just called `_opt` rather than `_or_nullopt`

Reply:
changed